cmd/erasure-object.go

```diff
@@ -1299,3 +1295,124 @@ func (er erasureObjects) GetObjectTags(ctx context.Context, bucket, object strin
 
 	return tags.ParseObjectTags(oi.UserTags)
 }
+
+// TransitionObject - transition object content to target tier.
+func (er erasureObjects) TransitionObject(ctx context.Context, bucket, object string, opts ObjectOptions) error {
+	tgtClient, err := globalTierConfigMgr.getDriver(opts.Transition.Tier)
+	if err != nil {
+		return err
+	}
+	// Acquire a write lock before deleting the object.
+	lk := er.NewNSLock(bucket, object)
+	ctx, err = lk.GetLock(ctx, globalDeleteOperationTimeout)
+	if err != nil {
+		return err
+	}
+	defer lk.Unlock()
+
+	fi, metaArr, onlineDisks, err := er.getObjectFileInfo(ctx, bucket, object, opts, false)
```

Comment:
You are likely to need the data.
```suggestion
	fi, metaArr, onlineDisks, err := er.getObjectFileInfo(ctx, bucket, object, opts, true)
```

Reply:
(it will be required for #11758 anyway)